scripts/diff-pkg-changes.py

```diff
@@ -0,0 +1,351 @@
+GITHUB_CONF_FNAME: str = 'conf/distro/nilrt.conf'
+
+def main(args: Namespace) -> int:
+    # Get user input
+    old_branch: str = args.old_branch
+    new_branch: str = args.new_branch
+    feed_base: str = args.feed_base
+
+    # Get the feed version
+    old_feed: str = get_feed(old_branch)
+    new_feed: str = get_feed(new_branch)
+
+    # Get the packages
+    old_core_pkgs: list['Package'] = get_packages(feed_base, old_feed, 'main', old_branch)
+    new_core_pkgs: list['Package'] = get_packages(feed_base, new_feed, 'main', new_branch)
+    old_extra_pkgs: list['Package'] = get_packages(feed_base, old_feed, 'extra', old_branch)
+    new_extra_pkgs: list['Package'] = get_packages(feed_base, new_feed, 'extra', new_branch)
+
+    # Compare them and save the info
+    report: dict = gen_report(
+        old_core_pkgs, new_core_pkgs, old_extra_pkgs, new_extra_pkgs,
+        args.old_vers, args.new_vers
+    )
+
+    print(dumps(report))
+
+    return 0
+
+# Error handling base class for the logic
+class PkgDiffError(Exception):
+    pass
+
+# When a file within the branch cannot be found
+class BranchFileNotFoundError(PkgDiffError):
+    def __init__(self, branch_name: str, file_name: str) -> 'BranchFileNotFoundError':
+        self.branch_name = branch_name
+        self.file_name = file_name
+        super().__init__('Failed to find file \'%s\' in branch %s.' % (branch_name, file_name))
+
+# When there's a valid branch, but its feed doesn't exist
+class FeedDoesNotExistError(PkgDiffError):
+    def __init__(self, branch_name: str, feed_name: str) -> 'FeedNotFound':
+        self.branch_name = branch_name
+        super().__init__('Failed to find feed \'%s\' in branch %s.' % (feed_name, branch_name))
+
+# Some issue with the Makefile containing the feed name
+class InvalidBranchFileError(PkgDiffError):
+    def __init__(self, branch_name: str, file_name: str) -> 'InvalidBranchFile':
+        self.branch_name = branch_name
+        self.file_name = file_name
+        super().__init__(
+            'Information in file \'%s\' in branch \'%s\' contained unexpected information.'
+        )
+
+# For things like logic errors that shouldn't happen, but could with improper input
+class UnexpectedError(PkgDiffError):
+    def __init__(self, msg):
+        self.msg = msg
+        super().__init__(msg)
+
+# Store CLI arguments here to make it clearer in other parts of the code
+class CLIParser(ArgumentParser):
+    def __init__(self) -> 'CLIParser':
+        super().__init__(
+            description = 'Diff the package feeds of two NI Linux RT releases via GitHub branches.'
+        )
+        self.add_argument(
+            'old_branch',
+            metavar = 'OLD', type = str,
+            help = 'The nilrt branch of the older release'
+        )
+        self.add_argument(
+            'old_vers',
+            metavar = 'OLD_ID', type = str,
+            help = 'The version number of the older release'
```

Comment:
I'm not sure why it would be confusing. I think it is more work for the user to look up those versions and pass it to the script. Let's talk offline to see what I'm missing.

Reply:
I think it would just be
```
if line.startswith('DISTRO_VERSION = '):
    distro_line = line
    break
``